src/v/redpanda/admin/api-doc/transform.json

```diff
@@ -0,0 +1,155 @@
+{
+  "apiVersion": "0.0.1",
+  "swaggerVersion": "1.2",
+  "basePath": "/v1",
+  "resourcePath": "/transform",
+  "apis": [
+    {
+      "path": "/v1/transform/deploy",
+      "operations": [
+        {
+          "method": "POST",
+          "summary": "Deploy a transform based on the provided metadata and wasm binary in the request payload. The payload should be a json `deploy_transform_request` object immediately followed by the wasm binary without any delimiters.",
+          "nickname": "deploy_transform",
+          "consumes": [
+            "application/json+wasm"
+          ],
+          "type": "void",
+          "produces": [
+            "application/json"
+          ]
+        }
+      ]
+    },
+    {
+      "path": "/v1/transform/delete",
+      "operations": [
+        {
+          "method": "POST",
+          "summary": "Delete a transform based on the provided name in the request payload.",
+          "type": "void",
+          "nickname": "delete_transform",
+          "consumes": [
+            "application/json"
+          ],
+          "produces": [
+            "application/json"
+          ]
+        }
+      ]
+    },
+    {
+      "path": "/v1/transform/list",
+      "operations": [
+        {
+          "method": "GET",
+          "summary": "List all transforms, the status report contains a cluster wide aggregated view of the live transform execution state.",
+          "nickname": "list_transforms",
+          "produces": [
+            "application/json"
+          ]
+        }
+      ]
+    }
+  ],
+  "models": {
+    "deploy_transform_request": {
+      "id": "deploy_transform_request",
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "input_topic": {
+          "type": "string"
+        },
+        "output_topics": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "environment": {
+          "type": "array",
+          "items": {
+            "type": "environment_variable"
```

Comment:
```suggestion
            "type": "object"
```
The `environment_variable` type is simply a key-value pair. So couldn't this be typed to a JSON object?

Reply:
Is there a way to enforce the values are strings only?

Reply:
The Admin server uses rapidjson and the method to get a string from the value in a json object is `value.GetString()`.
The `GetString()` method returns a string type. Take a look at `patch_cluster_config_handler` in `admin_server.cc` for an example.

Reply:
If you call `GetString` on something that isn't a string there is an `assert` that fires and it will crash the process. That doesn't enforce/validate anything.

You can work around that by using the json schema validators, but I'm hoping to avoid that by using the code generator here. Is there a recommended pattern? I haven't seen anything.

Reply:
No recommended pattern that I know of. Using a custom type is not a problem anyways, this looks good.